### PR TITLE
Production Sandbox separation for Websocket API

### PR DESCRIPTION
**Pull request by Arshardh**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java [hunk 1]
```diff
@@ -251,7 +252,10 @@ public synchronized static APIGatewayManager getInstance() {
                         }
                     } else {
                         String fileName = api.getContext().substring(1).replace('/', '-');
-                        client.undeployWSApi(new String[] { fileName });
+                        String[] fileNames = new String[2];
+                        fileNames[0] = "_PRODUCTION_" + fileName;
+                        fileNames[1] = "_SANDBOX_" + fileName;
```
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java]: Use constants here

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java [hunk 2]
```diff
@@ -279,36 +283,54 @@ public synchronized static APIGatewayManager getInstance() {
...
+                String msg = "Error while parsing the policy to get the eligibility query: ";
+                log.error(msg, e);
+                throw new APIManagementException(msg);
+            } catch (IOException e) {
+                String msg = "Error while deploying the policy in gateway manager: ";
+                log.error(msg, e);
+                throw new APIManagementException(msg);
+            }
+        } catch (JSONException e) {
+            log.error("Error in reading JSON object " + e.getMessage(), e);
```
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java]: Why the exception is not thrown?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java (hunk 1 above)
Comment (Arshardh) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java]: Fixed in 	56028bc

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java (hunk 2 above)
Comment (Arshardh) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java]: Fixed in 	56028bc

**Pull request merged**